core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -257,7 +257,7 @@ object DumpLogSegments {
     private def parseGroupMetadata(groupMetadataKey: GroupMetadataKey, payload: ByteBuffer) = {
       val groupId = groupMetadataKey.key
       val group = GroupMetadataManager.readGroupMessageValue(groupId, payload)
-      val protocolType = group.protocolType
+      val protocolType = group.protocolType.getOrElse("NONE")
```

Comment:
We could I suppose. It will only be `None` when the group is empty, so maybe I can skip iterating over the members in that case.